### PR TITLE
Optimization 4427 v1

### PR DESCRIPTION
**Pull request by jufajardini**
Make sure these boxes are signed before submitting your Pull Request -- thank you.

- [x] I have read the contributing guide lines at https://redmine.openinfosecfoundation.org/projects/suricata/wiki/Contributing
- [x] I have signed the Open Information Security Foundation contribution agreement at https://suricata-ids.org/about/contribution-agreement/
- [ ] I have updated the user guide (in doc/userguide/) to reflect the changes made (if applicable)

Link to [redmine](https://redmine.openinfosecfoundation.org/projects/suricata/issues) ticket:
https://redmine.openinfosecfoundation.org/issues/4427

Previous PR:
- https://github.com/OISF/suricata/pull/6053

Describe changes:
- ebpf/util: use dedicated 'id' type for flow storage as well
- device/storage: use dedicated 'id' type
- host/storage: use dedicated 'id' type
- ippair/storage: use dedicated 'id' type

#suricata-verify-pr:
#suricata-verify-repo:
#suricata-verify-branch:
#suricata-update-pr:
#suricata-update-repo:
#suricata-update-branch:
#libhtp-pr:
#libhtp-repo:
#libhtp-branch:

### CODE REVIEW

File: src/detect-engine-tag.c [hunk 1]
```diff
@@ -117,7 +117,7 @@ static DetectTagDataEntry *DetectTagDataCopy(DetectTagDataEntry *dtd)
  * \param p pointer to the current packet
  * \param tde pointer to the new DetectTagDataEntry
  *
- * \retval 0 if the tde was added succesfuly
+ * \retval 0 if the tde was added successfully
```
Comment (jlucovsky) [inline src/detect-engine-tag.c:120]: Suggest spelling/typos go into a separate "typo" commit.

File: src/ippair-storage.c [hunk 2]
```diff
@@ -58,8 +58,13 @@ void IPPairFreeStorage(IPPair *h)
         StorageFreeAll((Storage *)((void *)h + sizeof(IPPair)), STORAGE_IPPAIR);
 }
 
-int IPPairStorageRegister(const char *name, const unsigned int size, void *(*Alloc)(unsigned int), void (*Free)(void *)) {
-    return StorageRegister(STORAGE_IPPAIR, name, size, Alloc, Free);
+IPPairStorageId IPPairStorageRegister(const char *name, const unsigned int size,
+        void *(*Alloc)(unsigned int), void (*Free)(void *))
+{
+    // return StorageRegister(STORAGE_IPPAIR, name, size, Alloc, Free);
```
Comment (jlucovsky) [inline src/ippair-storage.c:64]: Delete commented out line

File: src/ippair-bit.c [hunk 3]
```diff
@@ -38,7 +38,7 @@
 #include "util-unittest.h"
 #include "ippair-storage.h"
 
-static int ippair_bit_id = -1;                /**< IPPair storage id for bits */
+static IPPairStorageId ippair_bit_id = { .id = -1 }; /**< IPPair storage id for bits */
```
Comment (jlucovsky) [inline src/ippair-bit.c:41]: Preexisting but suggest remaining this var to add a prefix, e.g. `g_ippair_bit_id` or perhaps `g_ippair_bit_storage_id`

**Comment by jufajardini** [comment]
Followed by: https://github.com/OISF/suricata/pull/6057

**Pull request closed without merging**